### PR TITLE
Populate README and remove duplicate

### LINKED ISSUE #26: Remove duplicate README
There are currently two files with the filename `README` in the root of the repository, one with `.txt` and one with `.md` as extensions. We should remove one, as having two in the repository root is confusing and only one gets rendered on the library homepage.

I prefer keeping the markdown file, since it means we can have rich text formatting unlike with pure text and also that file is older (9 years according to the commit history).

### LINKED ISSUE #27: Flesh out README with useful information and showcase of plots
The README is very bare at the moment. We should populate it with plenty of information, for example quoting from [the GitHub documentation page on READMEs](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-readmes):

>  A README is often the first item a visitor will see when visiting your repository. README files typically include information on:
> 
> What the project does
> Why the project is useful
> How users can get started with the project
> Where users can get help with your project
> Who maintains and contributes to the project

In particular I would like to include some images and links, notably showcasing some nice plots that cf-plot can make, and linking and advertising cf-python as the underlying data analysis tool.

### PR DESCRIPTION
**Pull request by sadielbartholomew**
Close #26 and close #27, by changing the codebase so there is only a single README document, in markdown format, with all the various details one would expect to find in a README and with lots of links to further resources. This can of course be tweaked in future but I wanted to flesh it out with plenty of relevant information as a first go.

@davidhassell I would appreciate it you could have a quick read through and let me know your thoughts as to whether I've missed anything or if any wording should be changed, or anything moved around, and so on.  (FYI until I move the documentation over from Andy's userspace I am putting any media resources e.g. images in a `docs/media` directory.) Thanks.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,86 @@
...
+It is designed to be a useful visualisation tool for environmental, earth
+and aligned science, for example to facilitate climate and meteorological
+research. cf-plot is developed and maintained by the
+[NCAS-CMS](https://cms.ncas.ac.uk/index.html) group, part of
+[NCAS](https://ncas.ac.uk/).
 
+
+### Brief Demonstration
+
+In as little as four lines of Python including imports, using `cf-plot` you can
```
Comment (davidhassell) [inline README.md:23]:
```suggestion
In as little as four lines of Python including, imports and file reading, using `cf-plot` you can
```

File: README.md [hunk 2]
```diff
@@ -1,13 +1,86 @@
...
+```
+
+
+### Examples Gallery
+
+A gallery of outputs made with cf-plot, showcasing a range of plotting
+possibilities with links to relevant documentation pages and to example code,
+can be found
+[on this dedicated page within the documentation](http://ajheaps.github.io/cf-plot/gallery.html),
+as illustrated in this (static) image:
```
Comment (davidhassell) [inline README.md:40]:
```suggestion
The [gallery of outputs made with cf-plot](http://ajheaps.github.io/cf-plot/gallery.html) showcases a range of plotting
possibilities with links to relevant documentation pages and to example code,
as illustrated in this (static) image:
```

File: README.md [hunk 3]
```diff
@@ -1,13 +1,86 @@
...
+and align with future ambitions for the library.
+
+Contributing guidelines will be added to the repository shortly.
+
+
+### Help: Issues, Questions, Feature Requests, etc.
+
+For any queries, see the
+[guidance page](http://ajheaps.github.io/cf-plot/issues.html)
+(`http://ajheaps.github.io/cf-plot/issues.html`).
```
Comment (davidhassell) [inline README.md:86]: Surely https://github.com/NCAS-CMS/cf-plot/issues ?

File: README.md [hunk 4]
```diff
@@ -1,13 +1,86 @@
...
-Please refer to the cf-plot homepage http://ajheaps.github.io/cf-plot
+**cf-plot allows you to produce and customise publication-quality contour,
+vector, line and more plots with the power of Python,
+[matplotlib](https://matplotlib.org/),
+[Cartopy](https://scitools.org.uk/cartopy/docs/latest/) and
+[cf-python](https://ncas-cms.github.io/cf-python/), in as few lines of code
+as possible.**
 
+It is designed to be a useful visualisation tool for environmental, earth
+and aligned science, for example to facilitate climate and meteorological
```
Comment (davidhassell) [inline README.md:15]:
```suggestion
and aligned sciences, for example to facilitate climate and meteorological
```

File: README.md (hunk 2 above)
Comment (davidhassell) [reply README.md:40]: ... thinking more, maybe you could move the image to the top of the README, so that it gets seen by the don't-scroll-down crowd. Just a thought.

Comment (sadielbartholomew) [reply README.md:40]: I did wonder whether this would be best, but also wanted to highlight to *that* crowd how few lines are required to get a plot out and put that first. However, now you mention it, they are more likely to scroll further past an image, so I agree with your suggestion. Thanks, I'll promote it to the first section, then mark this comment as resolved.

File: README.md [hunk 5]
```diff
@@ -1,13 +1,86 @@
...
+pip install cf-python
+pip install cf-plot
+```
+
+or you can use `conda` (or similar package managers such
+as `mamba`) as follows (or equivalent):
+
+```bash
+conda install -c ncas -c conda-forge cf-python cf-plot udunits2
+conda install -c conda-forge mpich esmpy
```
[the remark this answers is not in the document]
Comment (sadielbartholomew) [reply README.md:65]: Oh fair. I was just copying what Andy has on the ['Download/Install' page](Download/Install) without thinking over it much. I'll remove the line here and make sure it is clearer on that page that the second line of installs is optional.

File: README.md [hunk 6]
```diff
@@ -1,13 +1,86 @@
...
+More detail about installation is provided on the
+[installation page](http://ajheaps.github.io/cf-plot/download.html)
+(`http://ajheaps.github.io/cf-plot/download.html`)
+of the documentation.
+
+### Contributing
+
+Everyone is welcome to contribute to cf-plot, however the NCAS-CMS team reserve
+the right to only accept changes which they consider to improve the codebase
+and align with future ambitions for the library.
```
[the remark this answers is not in the document]
Comment (sadielbartholomew) [reply README.md:77]: Indeed, I just thought it might be wise to indicate that we can't be bound to accepting any old changes that someone might propose. Notably, in a previous project I worked on, someone did months of work towards a PR without checking in (formally/regularly) that it was what was wanted by the core project team, and after difficult conversations the PR was closed and not incorporated and I imagine someone felt very disgruntled. I think mainly I wanted to indicate that any work contributed should be aligned with an Issue raised and signed off by myself or someone else relevant in NCAS-CMS, but probably if we say something along those lines it should be in the more detailed Contributing guidance I will soon write up (or specifically, copy from another good OS project).

In short, I agree with your proposal but think we should add something along the above lines to the more detailed guidance, to prevent anyone possibly doing some work towards contributions which we can't accept.

Committing your suggestion (there's a typo but I'll fix that up in a final commit). Thanks.

File: README.md (hunk 3 above)
Comment (sadielbartholomew) [reply README.md:84]: I haven't yet moved the documentation source and build across to the NCAS-CMS space, so technically that is the canonical link at present, though Andy has set up redirection for now. I was planning to update the links once I had done the move, but you're right it might be best just put the eventual ones in for now (will be less confusing to refer to the same user/org space as the repo belongs to). Thanks.

File: README.md (hunk 2 above)
Comment (sadielbartholomew) [reply README.md:40]: Actually, re-reading I see you were probably instead proposing to move the image only, not that whole section. Might be best, good thinking. I'll do that and point to it from the 'Examples Gallery' section, left in the same place.

File: README.md (hunk 3 above)
Comment (sadielbartholomew) [reply README.md:84]: Oh, realising I misinterpreted what you said in my reply above (ignore it).

Andy's current guidance page states people should email me directly. I have started to initiate the Issue Tracker for this instead, but haven't had a chance to update the guidance. Until I update that I will leave this pointer so as not to make contradictory statements RE issue reporting.

Comment (sadielbartholomew) [reply README.md:85]: Hope that makes sense and is OK to you. I'll leave this unresolved in case you want to say otherwise :slightly_smiling_face:

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (sadielbartholomew) [reply README.md]: See [3cb767e](https://github.com/NCAS-CMS/cf-plot/pull/30/commits/3cb767e65d6d69c86312b03d814091a1a5870e51).

File: README.md (hunk 2 above)
Comment (sadielbartholomew) [reply README.md]: See [3cb767e](https://github.com/NCAS-CMS/cf-plot/pull/30/commits/3cb767e65d6d69c86312b03d814091a1a5870e51) where I've moved the image (only) to the top.

**Comment by sadielbartholomew** [comment]
Thanks for all of your very useful comments, @davidhassell. I believe I have addressed everything, so let me know if you are now happy to merge.

**Comment by davidhassell** [comment]
All good! Please merge ...
